overlord/devicestate/devicemgr.go

```diff
@@ -172,12 +183,94 @@ func (m *DeviceManager) ensureOperational() error {
 	return nil
 }
 
+var bootPopulateStateFromSeed = boot.PopulateStateFromSeed
+
+// ensureSnaps makes sure that the snaps from seed.yaml get installed
+// with the matching assertions
+func (m *DeviceManager) ensureSeedYaml() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	// FIXME: enable on classic?
+	//
+	// Disable seed.yaml on classic for now. In the long run we want
+	// classic to have a seed parsing as well so that we can install
+	// snaps in a classic environment (LP: #1609903). However right
+	// now it is under heavy development so until the dust
+	// settles we disable it.
+	if release.OnClassic {
+		return nil
+	}
+
+	var seeded bool
+	err := m.state.Get("seeded", &seeded)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+	if seeded {
+		return nil
+	}
+
+	if m.changeInFlight("seed") {
+		return nil
+	}
+
+	tsAll, err := bootPopulateStateFromSeed(m.state)
+	if err != nil {
+		return err
+	}
+	if len(tsAll) == 0 {
+		return nil
+	}
+
+	msg := fmt.Sprintf("Initialize system state")
+	chg := m.state.NewChange("seed", msg)
+	for _, ts := range tsAll {
+		chg.AddAll(ts)
+	}
+	ts := tsAll[len(tsAll)-1]
+	markSeeded := m.state.NewTask("mark-seeded", i18n.G("Mark system seeded"))
+	markSeeded.WaitAll(ts)
+	chg.AddTask(markSeeded)
+
+	m.state.EnsureBefore(0)
+
+	return nil
+}
+
+func (m *DeviceManager) ensureBootOk() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	if release.OnClassic {
+		logger.Debugf("Ignoring 'booted' on classic")
+		return nil
+	}
+
+	bootloader, err := partition.FindBootloader()
+	if err != nil {
+		return fmt.Errorf(i18n.G("cannot mark boot successful: %s"), err)
+	}
+
+	if err := partition.MarkBootSuccessful(bootloader); err != nil {
```

Comment:
Excellent point, it will run this every 5min. It won't actually write the bootloader bits every 5min because it checks internally if the var has actually changed. However its still silly, so added a flag.